Kconfig.zephyr

```diff
@@ -448,6 +448,7 @@ config CODING_GUIDELINE_CHECK
 config NATIVE_LIBC
 	bool
 	select FULL_LIBC_SUPPORTED
+	select TC_PROVIDES_POSIX_C_LANG_SUPPORT_R
```

Comment:
So previously we didn't provide POSIX reentrant functions in case of native build? Or we provided implementation from Zephyr?

Reply:
Yes ? Hard to say - I don't often test on native_sim

Reply:
> So previously we didn't provide POSIX reentrant functions in case of native build? Or we provided implementation from Zephyr?

It's complicated. The native implementation always provided the functions but our configuration system was not aware of the fact. It kinda worked, because in-tree users don't bother with selecting the right symbol anyway. But as soon as the libc test got its dependencies fixed, the issue was uncovered.